configure-freebsd.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/local/bin/bash
+
```

Comment:
Since I know very little about FreeBSD, here are a few more questions:

- What is the standard libpcap version for FreeBSD? I'm asking to see if libpcap's "immediate mode" is supported
- What is the default install dir? is it `/use/local` like in Linux and MacOS? Did you try the install script?
- Did you test any PcapPlusPlus example that excepts live traffic? Does it work properly?

Reply:
1. It depends on FreeBSD release you installed. I'd suggest you to only support FreeBSD release which is no in EOL. Then the current FreeBSD is 11.2 or above. The libpcap in 11.2 is `libpcap-1.9.0`.
1. `/usr/local` is your only choice. The install script failed because there is only one template. You need to replace `#!/bin/bash` to `#!/usr/local/bin/bash` and replace `sed` to `gsed`. TBH, system admin rarely install app like this. They prefer to use FreeBSD ports. That's why I mention that we should submit a FreeBSD port file in upstream, instead.
1. Yes. I tested live capture. It works. That's why I fixed the default time out value in live capture. FreeBSD is more like Mac OS X.

Reply:
For bullet point 2, I added commit 92b53d12d which use a FreeBSD version `install.sh.freebsd.template`.

Reply:
Thanks for adding the installation file! We can definitely submit a FreeBSD port request, do you know what is the process to do it?

Reply:
No, I have never submit a FreeBSD port before.

I sent a PR to patch a port maintained by pfsense. I learnt the port creation from [the FreeBSD porter handbook](https://www.freebsd.org/doc/en_US.ISO8859-1/books/porters-handbook/). FreeBSD has the best documentation among all *nix OS. It should be easy peasy. We will definitely do it later.

If you are OK with everything, let's merge this PR. I will start work on my research project.

BTW, you may overlook my comment about Cirrus CI. It is done in my another dev branch. Check it out my comment in the issue #260.

Reply:
I agree, let's get the PR merged first and then think about FreeBSD port.

Regarding Cirrus CI - can you please add `.cirrus.yml` to the PR? once it's merged I'll enable CirrusCI as well. Please also add the unit-tests to the CI job (now that it's fixed)

Reply:
There is still an issue in pcap++test. I get a core dump in raw socket test. You may need help from other to fix raw socket.

I will add the CI if you can confirm [the patch](https://github.com/seladb/PcapPlusPlus/pull/261#discussion_r338332888) looks ok to you.